assets/js/router.js

```diff
@@ -0,0 +1,25 @@
+import { router, navigate } from "https://unpkg.com/million@1.8.9-0/dist/router.mjs"
+
+export const init = (loader) => {
+  // SPA navigation for access later
+  window.navigate = navigate
+  // We only mutate document.title and content within .singlePage element
+  router(".singlePage")
+  // We need on initial load, then subsequent redirs
+  window.addEventListener("million:navigate", () => callback(loader))
+  window.addEventListener("DOMContentLoaded", () => callback(loader))
+}
+
+export const callback = (loader) => {
+  // requestAnimationFrame() delays graph draw until SPA routing is finished
+  const draw = () => {
+    const container = document.getElementById("graph-container")
+    // retry if the graph is not ready
+    if (!container) return requestAnimationFrame(draw)
+    // clear the graph in case there is anything within it
+    container.textContent = ""
```

Comment:
should this be in the callback that is passed to `init` in `head.html#L70`? ideally consolidate all the re-rendering logic in one place

Reply:
Fixed in [d38f9be](https://github.com/jackyzha0/quartz/pull/123/commits/d38f9bec70c7e46d9e7662ba3925abc9cdd9370a)